src/devdocs2zim/generator.py

```diff
@@ -0,0 +1,281 @@
+
+from pydantic import BaseModel
+from zimscraperlib.constants import (  # pyright: ignore[reportMissingTypeStubs]
+    MAXIMUM_DESCRIPTION_METADATA_LENGTH,
+    MAXIMUM_LONG_DESCRIPTION_METADATA_LENGTH,
+    RECOMMENDED_MAX_TITLE_LENGTH,
+)
+
+from devdocs2zim.client import (
+    DevdocsClient,
+    DevdocsMetadata,
+)
+from devdocs2zim.constants import logger
+
+
+class InvalidFormatError(Exception):
+    """Raised when a user supplied template has an invalid parameter."""
+
+    pass
+
+
+class ZimConfig(BaseModel):
+    """Common configuration for building ZIM files."""
+
+    # Directory the ZIM will be written to.
+    output_folder: str
+    # File name/name for the ZIM.
+    name_format: str
+    # Human readable title for the ZIM.
+    title_format: str
+    # Publisher for the ZIM.
+    publisher: str
+    # Creator of the content in the ZIM.
+    creator: str
+    # Short description for the ZIM.
+    description_format: str
+    # Long description for the ZIM.
+    long_description_format: str
+    # Semicolon delimited list of tags to apply to the ZIM.
+    tags: str
+
+    @staticmethod
+    def add_flags(parser: argparse.ArgumentParser, defaults: "ZimConfig"):
+        """Adds flags to the given parser with the given defaults."""
+        parser.add_argument(
+            "--output",
+            help=f"Output folder for ZIMs. Default: {defaults.output_folder}",
+            default=defaults.output_folder,
+            dest="output_folder",
+        )
+
+        parser.add_argument(
+            "--creator",
+            help=f"Name of content creator. Default: {defaults.creator!r}",
+            default=defaults.creator,
+        )
+
+        parser.add_argument(
+            "--publisher",
+            help=f"Custom publisher name. Default: {defaults.publisher!r}",
+            default=defaults.publisher,
+        )
+
+        parser.add_argument(
+            "--name-format",
+            help="Custom name format for individual ZIMs. "
+            f"Default: {defaults.name_format!r}",
+            default=defaults.name_format,
+            metavar="FORMAT",
+        )
+
+        parser.add_argument(
+            "--title-format",
+            help=f"Custom title format for individual ZIMs. "
+            f"Value will be truncated to {RECOMMENDED_MAX_TITLE_LENGTH} chars. "
+            f"Default: {defaults.title_format!r}",
+            default=defaults.title_format,
+            metavar="FORMAT",
+        )
+
+        parser.add_argument(
+            "--description-format",
+            help="Custom description format for individual ZIMs. Value will be "
+            f"truncated to {MAXIMUM_DESCRIPTION_METADATA_LENGTH} chars. "
+            f"Default: {defaults.title_format!r}",
+            default=defaults.description_format,
+            metavar="FORMAT",
+        )
+
+        parser.add_argument(
+            "--long-description-format",
+            help="Custom long description foramt for your ZIM. Value will be "
```

Comment:
Typo `foramt` => `format`

Reply:
fixed, also found this in the README and fixed that